### PR TITLE
Replace XML descriptor with DynamicDependency for Array special methods

### PR DESCRIPTION
**Pull request by marek-safar**
to allow better trimming of unused methods

Part of #44859

### CODE REVIEW

**Review by vargaz — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.FileProviders.Composite/tests/MockFileProvider.cs [hunk 1]
```diff
@@ -11,6 +11,13 @@ namespace Microsoft.Extensions.FileProviders.Composite
 {
     public class MockFileProvider : IFileProvider
     {
+#if NETCOREAPP
+        static MockFileProvider ()
```
Comment (jkotas) [inline src/libraries/Microsoft.Extensions.FileProviders.Composite/tests/MockFileProvider.cs:15]:
```suggestion
        static MockFileProvider()
```

File: src/libraries/Microsoft.Extensions.FileProviders.Composite/tests/MockFileProvider.cs [hunk 2]
```diff
@@ -11,6 +11,13 @@ namespace Microsoft.Extensions.FileProviders.Composite
...
+        {
+            Castle.DynamicProxy.Generators.AttributesToAvoidReplicating.Add(typeof(System.Diagnostics.CodeAnalysis.DynamicDependencyAttribute));
```
Comment (jkotas) [inline src/libraries/Microsoft.Extensions.FileProviders.Composite/tests/MockFileProvider.cs:17]: Why is this needed? It may be useful to add comment.

Comment (marek-safar) [reply src/libraries/Microsoft.Extensions.FileProviders.Composite/tests/MockFileProvider.cs:19]: Comment added

**Review by jkotas — APPROVED** [review]
Thanks!
